### PR TITLE
chore: Add github action for stress tests in all categories

### PR DESCRIPTION
**Pull request by thisisabhash**
*Issue #, if available:* None 

*Description of changes:* 
Add github action for stress tests in all categories. 
All categories except GraphQLAPI use the same configuration file (all categories are added to a single Amplify app).

**Note: GraphQLAPI stress tests need a separate configuration file as conflict detection needs to be turned on for Datastore Stress tests.**

*Check points: (check or cross out if not relevant)*

- [ ] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [x] All integration tests pass
- [x] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [x] PR title conforms to conventional commit style
- [ ] If breaking change, documentation/changelog update with migration instructions

*DataStore checkpoints (check when completed)*

- [ ] Ran AWSDataStorePluginIntegrationTests
- [ ] Ran AWSDataStorePluginV2Tests
- [ ] Ran AWSDataStorePluginMultiAuthTests
- [ ] Ran AWSDataStorePluginCPKTests
- [ ] Ran AWSDataStorePluginAuthCognitoTests
- [ ] Ran AWSDataStorePluginAuthIAMTests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by thisisabhash** [comment]
Stress Test run successful - https://github.com/aws-amplify/amplify-swift/actions/workflows/stress_test.yml

**Review by royjit — DISMISSED** [review]
LGTM

File: AmplifyPlugins/DataStore/Tests/DataStoreHostApp/DataStoreStressTests/DataStoreStressTests.swift [hunk 1]
```diff
@@ -48,13 +48,15 @@ final class DataStoreStressTests: DataStoreStressBaseTest {
             posts.append(post)
         }
         
+        let observeQuerySetup = asyncExpectation(description: "ObserveQuery setup done")
```
Comment (lawmicha) [inline AmplifyPlugins/DataStore/Tests/DataStoreHostApp/DataStoreStressTests/DataStoreStressTests.swift:51]: why is this needed?

Comment (thisisabhash) [reply AmplifyPlugins/DataStore/Tests/DataStoreHostApp/DataStoreStressTests/DataStoreStressTests.swift:51]: datastore stress tests were failing due to timeout intermittently. i figured that observeQuery was not being setup at the right time since it starts a new task. after this change, the tests are running reliably.

Comment (thisisabhash) [reply AmplifyPlugins/DataStore/Tests/DataStoreHostApp/DataStoreStressTests/DataStoreStressTests.swift]: moved observe API out of the task

**Review by lawmicha — APPROVED** [review]

**Review by royjit — APPROVED** [review]
